### PR TITLE
adding hyperbolic tangent and tests

### PR DESCRIPTION
**Pull request by deckerla**
added hyperbolic tangent operator for real numbers and associated test

### CODE REVIEW

File: src/jop_tanh.jl [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+where `F` is the hyperbolic tangent operator with domain and range given by `spc::JetSpace`.
+we use 'tanh'(x) = (\exp{x}-\exp{-x})/(\exp{x}+\exp{-x}),
+the derivative of which is 1-tanh(x)^2.
+We expect the domain and range to be real.
+"""
+JopTanh(spc::JetSpace) = JopNl(dom = spc, rng = spc, f! = JopTanh_f!, df! = JopTanh_df!)
+export JopTanh
+
+# TODO: assert denominator in linearization is > 0
```
Comment (samtkaplan) [inline src/jop_tanh.jl:12]: did you mean to include this comment? the linearization does not seem to have a denominator.

File: test/jop_tanh.jl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    observed, expected = linearization_test(F, m0, μ = μ, δm = δm)
+    δ = minimum(abs, observed - expected)
+    @test δ < 0.1
+end
+
+@testset "JopTanh, odd funcion T=$(T)" for T in (Float64,Float32)
+    F = JopTanh(JetSpace(T,n1,n2))
+    x1 = 2 .* (-1 .+ 2 .* rand(domain(F)))
+    @test F * (-1 .* x1) ≈ -(F * x1)
+end
```
Comment (samtkaplan) [inline test/jop_tanh.jl:40]: I guess we should add new lines at the end of files.

**Review by samtkaplan — CHANGES_REQUESTED** [review]
I'm curious, does this work for complex?

**Comment by deckerla** [comment]
added complex number functionality and resolved issues outlined above

**Review by samtkaplan — APPROVED** [review]

**Pull request merged**